### PR TITLE
Rule S1994: for loop increment clauses should modify the loops' counters

### LINKED ISSUE #63: Rule S1994: "for" loop increment clauses should modify the loops' counters
Implements [RSPEC-1994](https://jira.sonarsource.com/browse/RSPEC-1994)

### PR DESCRIPTION
**Pull request by vilchik-elena**
fixes #63

### CODE REVIEW

File: common-rule-engine/src/main/java/org/sonar/commonruleengine/checks/ForLoopConditionVariableCheck.java [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+  private static final String MESSAGE = "This loop's stop condition tests variable which is not incremented in update clause.";
+
+  public ForLoopConditionVariableCheck() {
+    super(Kind.FOR);
+  }
+
+  @Override
+  public void visitNode(UastNode node) {
+    Optional<UastNode> optionalUpdateClause = node.getChild(Kind.FOR_UPDATE);
+    Optional<UastNode> optionalConditionClause = node.getChild(Kind.CONDITION);
```
Comment (saberduck) [inline common-rule-engine/src/main/java/org/sonar/commonruleengine/checks/ForLoopConditionVariableCheck.java:41]: I think we should look only for condition which is binary expression with relational or equality operator, to exclude loops which are doing some more sophisticated stuff

**Review by saberduck — CHANGES_REQUESTED** [review]

**Comment by saberduck** [comment]
@vilchik-elena seems that RSPEC is not updated to reflect the approach taken in the implementation

File: common-rule-engine/src/main/java/org/sonar/commonruleengine/checks/ForLoopConditionVariableCheck.java (hunk 1 above)
Comment (vilchik-elena) [reply common-rule-engine/src/main/java/org/sonar/commonruleengine/checks/ForLoopConditionVariableCheck.java]: @saberduck wdyt if I will keep only conditions with single identifier?

**Comment by vilchik-elena** [comment]
@saberduck indeed, as for any other language:) I will see what I can do

**Review by saberduck — APPROVED** [review]

**Comment by saberduck** [comment]
@vilchik-elena lgtm, rebase once master is green and we are good to go

**Pull request merged**